### ISSUE TITLE
Text overlay

### ISSUE THREAD

**Issue by thompson318** [UCL/scikit-surgeryvtk#10]
In GitLab by @ThomasDowrick on Dec 3, 2018, 18:01

So far I have implemented:

* Class to wrap vtkTextActor.
* Example pipeline of how to add text to the vtk window in response to a mouse click.
* Resizing of text if the window is resized.
* Class to wrap VTKCornerAnnotation and example usage. This class may not be necssary, as the underlying implementation is fairly simple, but I thought it better to make it and get rid of it later if not needed.

It still needs refactoring/further work, but I have committed what I have done for now, in case I don't get a chance to work on it again before paternity leave.

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Dec 3, 2018, 18:02

created branch [`10-text-overlay`](https://weisslab.cs.ucl.ac.uk/WEISS/SoftwareRepositories/SNAPPY/scikit-surgeryoverlay/compare/master...10-text-overlay)

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Dec 3, 2018, 18:03

mentioned in merge request !1

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Dec 13, 2018, 09:21

assigned to @MattClarkson

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Dec 13, 2018, 09:21

@ThomasDowrick - just in case you read this. Under Issue #11 I did some refactorings, and got unit tests to work, and tidied up a bit. 

There were a few design changes, as I want the overlay widget to be completely separate from any VideoSource. It wasn't a major change, and unit tests pass. 

I've also got a good idea (I hope) how i will do output to stereo nicely.

So, Im going to go through this branch and cherry pick the relevant bits, and integrate them into my new model. So I'll take this ticket. 

Long story short: Don't continue on this branch, as its probably obsolete.

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Dec 21, 2018, 23:12

It's taken me ages to get even close to getting all unit tests to work, so I will have to hand this back to you.

Essentially, by the time you read this, the master branch will have moved on a lot. So dont try to merge this branch at all.

I do suspect however that most of the actual stuff to do with displaying text will be ok.

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Dec 21, 2018, 23:12

assigned to @ThomasDowrick and unassigned @MattClarkson

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Feb 5, 2019, 07:56

Hi Tom, any free time to finish this one off?

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 5, 2019, 07:58

A few bits more to do on ardavin (should finish today), and then I'll move to this one.

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Feb 5, 2019, 08:12

Cool. Look forward to a demo of sorts on Thurs. Then we should talk about actually doing it in a DaVinci!

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 7, 2019, 10:54

mentioned in commit 6a754f42c718ad7f58b70bc943484e999beb9a0f

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 7, 2019, 10:54

mentioned in commit 22afd91c849e35e9881301d8e511193c5998f3f9

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 7, 2019, 16:45

mentioned in commit e522703665fc3a858aedd2c94183c09f14b8c0a7

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 7, 2019, 16:45

mentioned in commit 32fd2eea85ac8867ba5c67050a4182131ce6d020

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 7, 2019, 16:45

mentioned in commit 212cd56875e53150d538215dc4e845b9f1b66c97

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Feb 8, 2019, 13:25

i was thinking about this last night on train. 

We need to ensure that in the stereo widget, text is displayed at the same position in left and right window, so its not affected by the perspective geometry. Then on a 3D screen it will always appear at a constant depth. 

This could be achieved either by using a differnt layer with an independent camera, or by using vtkActor2D or whateer its callled.

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 8, 2019, 14:00

There is also an issue with maintaining the correct relative position of a vtkTextActor if the window is resized. I have this sort-of working by adding a callback to update the text position whenever the window dimensions change, but there may be a better solution.

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Feb 8, 2019, 14:20

in resizeEvent you would reposition the text actor.

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 8, 2019, 15:07

At the moment the actor listens for a VTK resize event, and then resizes. This is fairly equivalent to Qt widget moving it as part of resizeEvent, but I can have a think about which way is more sensible.

**Comment by thompson318** [comment]
In GitLab by @MattClarkson on Feb 8, 2019, 15:44

yeah, Im not sure which is best. I've preferred Qt, but thats out of familiarity more than anything else really.

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 10, 2019, 17:25

mentioned in commit 242128d54ce2d44563e4b9430d27ef4b5f2f1ac7

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 10, 2019, 17:25

mentioned in commit 2ab52d45f67533cdf97bea6d1eaa4fd87b8bfe8f

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 10, 2019, 17:25

mentioned in commit 0207551909a248533f35304f0c472d8aedb3a137

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 11, 2019, 15:08

mentioned in commit f1628291454847923bc9cca596faa53179df0004

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Feb 11, 2019, 15:08

mentioned in commit 102d2d62c641f4f05d05d6bd4db8dd9f786d53a4

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 13:20

mentioned in commit d470cf7bc73178d08e3736ecb0d97dde677cda61

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 13:29

mentioned in commit bfc2670235ffbc3660efdd13db4f5d061a1b38be

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 13:52

mentioned in commit f60a33fb65fec7ee955a4e550da40c3130e18291

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 16:22

mentioned in commit 4664a9a69acdf4c1f4034b5373790613a866bb1a

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 16:32

mentioned in commit 09c41b0e736acc172b02b7526e209ccd83b0090c

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 16:41

mentioned in commit fcefcb849e49ebc79f15220d242febdedd13bfbf

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 1, 2019, 16:49

mentioned in commit d5c73fee7a9c78054e393c85d8c2fb3143d30700

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 5, 2019, 11:16

mentioned in merge request !14

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 5, 2019, 11:34

mentioned in commit 171daff62901da77a8dc8ee443d5eed46dee609d

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 5, 2019, 11:45

mentioned in commit ec099d831d6b244a1b0378a19c798d5eeec68037

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 5, 2019, 11:45

mentioned in commit 0c02efa5ff2be59291f19b87e07fe5d9b1e89c2e

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 14, 2019, 10:02

mentioned in commit 3c021c0c221ee0acdac48f2db19fb69fb4082086

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 14, 2019, 12:54

mentioned in commit d6cfcc94ff020766d053e231e9c55d13a56e9b23

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 14, 2019, 12:54

mentioned in commit e454f6084f5c4e9af4282d698457c400f49819f0

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 20, 2019, 14:52

mentioned in commit 061fb0d4696272fb1d985e7c79f81504cf22b5c9

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 20, 2019, 16:09

mentioned in commit c8e59f5d3349dfc34474590f27c41bff9ddd32e8

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 21, 2019, 11:57

closed via merge request !14

**Comment by thompson318** [comment]
In GitLab by @ThomasDowrick on Mar 21, 2019, 11:57

mentioned in commit 6c6d371bc62776ee4ed5c310ed10021e182e72e1
